Source/Xamarin/Prism.Forms/Common/PageUtilities.cs

```diff
@@ -47,6 +47,7 @@ private static void DestroyChildren(Page page)
         {
             if (page is MasterDetailPage)
             {
+                DestroyPage(((MasterDetailPage)page).Master);
```

Comment:
I'm a fan of this being inside of `DestroyChildren`. The parent should be Destroyed by `InvokeViewAndViewModelAction<IDestructible>(page, v => v.Destroy());` on line 35. If that's not occurring I would like to know why and address that issue instead.